nixos/modules/tasks/network-interfaces.nix

```diff
@@ -305,6 +315,29 @@ let
       name = mkDefault name;
     };
 
+    # Renamed or removed options
+    imports = [
+      (mkRenamedOptionModule [ "ip4" ] [ "ipv4" "addresses"])
+      (mkRenamedOptionModule [ "ip6" ] [ "ipv6" "addresses"])
+      (mkRemovedOptionModule [ "subnetMask" ] ''
+        Supply a prefix length instead; use option
+        networking.interfaces.<name>.ipv{4,6}.addresses'')
+      (mkMergedOptionModule
+        [ [ "ipAddress" ] [ "prefixLength" ] ]
+        [ "ipv4" "addresses" ]
+        (cfg: with cfg;
+          lib.optional (ipAddress != null && prefixLength != null)
```

Comment:
There is a minor problem here. If `ipAddress` is set but `prefixLength` isn't, this still evaluates to true and gives a type error. `prefixLength` isn't `null` for some reason.

Reply:
prefixLength is the string `"_mkMergedOptionModule"` when not defined.

Reply:
Not sure if this is the intended usage but It works now.